sklearn/manifold/t_sne.py

```diff
@@ -521,5 +842,57 @@ def fit_transform(self, X, y=None):
         X_new : array, shape (n_samples, n_components)
             Embedding of the training data in low-dimensional space.
         """
-        self.fit(X)
+        embedding = self._fit(X)
+        self.embedding_ = embedding
         return self.embedding_
+
+    def fit(self, X):
```

Comment:
Can you maybe remove this method as was done in master? I find it confusing and it doesn't add anything.
Also, ``fit`` needs to return ``self``